reactome_analysis_datasets/reactome_analysis_datasets/dataset_fetchers/expression_atlas_fetcher.py

```diff
@@ -83,6 +88,50 @@ def get_dataset_id(self, parameters: list) -> str:
         """
         return self._get_parameter(name="dataset_id", parameters=parameters)
 
+    def get_available_datasets(self, no_datasets: int):
+        """
+        Returns the available datasets
+        :param no_datasets: number of datasets to retrieve
+        :returns: datasets in ExternalData format
+        """
+        experiments_external_data_list = list()
+        try:
+            experiments_url = "https://www.ebi.ac.uk/gxa/json/experiments"
+            response = requests.get(experiments_url)
+            response.raise_for_status()
+            json_response = response.json()
+            experiments_list = json_response['experiments'][0:no_datasets]
```

Comment:
Here I made a mistake. Similar to the grein_loader function the no_datasets should be changed to

```
def get_available_datasets(self, no_datasets: int=None): -> list
```

Can you adapt the AbstractDatasetFetcher as well?

You'd then have to adapt this line for cases where the no_datasets is not set.

Reply:
changed to a default value of 10000 datasets